Feature/d650fe40 Chart Finish

bh2980 opened the pull request:
차트 레이아웃 완료했습니다.
라우팅 시에 노션에 나온대로 /company에서 /detail로 바꿨습니다.
검토 부탁드립니다.

bh2980 committed:
feat(Company)  : change data to dummy except chart

bh2980 committed:
Merge branch 'feature/d650fe40' into develop

bh2980 committed:
Merge branch 'develop' into feature/d650fe40

bh2980 committed:
feat(Company) : divide price color class to red and blue

bh2980 committed:
fix(Company) : eslint parser error

bh2980 committed:
fix(Company) : vscode indent error

bh2980 committed:
feat(Company) : add chart dummy data

bh2980 committed:
Merge branch 'develop' into feature/d650fe40

bh2980 committed:
feat(Company) : Chart Edit

bh2980 committed:
fix(Company) : Routing domain change to /detail and edit layout

bh2980 committed:
feat(Company) : refect all dummy data to chart

x-axis not yet

bh2980 committed:
feat(Company) : Stock CHart End

bh2980 committed:
fix(Company) : line chart data deleted

kasterra commented on package.json:
```diff
@@ -8,6 +8,7 @@
     "@testing-library/user-event": "^13.2.1",
     "apexcharts": "^3.33.0",
     "node-sass": "^7.0.1",
+    "normalize-url": "^7.0.3",
```
다음부턴 패키지 깔 때 필요한거 있으면 말씀하세요. 여기다가 직접 깔지 말구...

bh2980 replied:
넵! 명심하겠습니다.
현재는 필요한 패키지는 아니고 저번에 에러 해결하려고 방법 찾아보다가 깐 거라서 제가 지우면 될 것 같습니다.
package.json이랑 yarn.lock 변경된 것도 같이 올리면 될까요?

kasterra replied:
네 당연하죠!

kasterra commented on .prettierrc:
```diff
@@ -5,7 +5,7 @@
     "bracketSpacing": true, 
     "semi": true,
-    "useTabs": true,
+    "useTabs": false,
```
.prettierc같이 모두가 영향받는 규칙 수정하실때도 모두에게... 물어보시고...

bh2980 replied:
넵... 인덴트 인코딩 에러 계속 나서 고쳐보다가 원복 안 한 것 같습니다. 지금은 다시 복구해놓았습니다.

bh2980 replied:
넵 죄송합니다. 인덴트 인코딩 에러가 계속 나서 해결하다가 원복을 안해놓은 것 같습니다.
지금은 다시 true로 돌려놓았습니다.

kasterra commented on src/components/Company/css/Company.scss:
```diff
@@ -115,17 +120,26 @@
-      margin: 0px 24px;
+      overflow: auto;
       font-size: 1.2em;
+      // -ms-overflow-style: none; /* IE and Edge */
```
? 코멘트 이거 안쓰시는거면 지워주세요

bh2980 replied:
반영했습니다.

kasterra commented on src/components/Company/css/Company.scss:
```diff
@@ -115,17 +120,26 @@
+      overflow: auto;
       font-size: 1.2em;
+      // -ms-overflow-style: none; /* IE and Edge */
+      // scrollbar-width: none; /* Firefox */
```
이것두요

bh2980 replied:
반영했습니다.

kasterra commented on .eslintrc:
```diff
@@ -1,13 +1,20 @@
 {
-    "extends": ["plugin:prettier/recommended"],
-    "parser": "babel-eslint",
```
babel-eslint를 날리신 이유가 있으신가요?

bh2980 replied:
babel은  저번에 에러 때문에 톡 드렸을 때 .eslintrc파일에 적어보라고 하셨던 거에서 아마 빠져있던 것 같습니다.
prettier는 왜...왜지...지금은 다시 넣어놓았습니다.
아 그리고 End of Line 에러 해결할 때 추가된 코드도 .eslintrc에 들어가있어서 확인 부탁드립니다.

kasterra reviewed (changes requested):
주석처리된 코드들은 날려주시고 다시 올려주세요~~

bh2980 committed:
Merge branch 'develop' into feature/d650fe40

kasterra commented on src/components/Company/Company.jsx:
```diff
@@ -1,101 +1,20 @@
-import React from "react";
-import "./Company.scss";
-import { ReactComponent as TitleIcon } from "assets/icons/company-title-icon.svg";
+import React, { useState } from 'react';
```
파일 제목 CompanyContainer로 바꿔주시고

bh2980 replied:
반영했습니다.

kasterra commented on src/components/Company/CompanyPresenter.jsx:
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
```
얘는 Company.jsx로 바꾸면 어떨까요?

bh2980 replied:
넵 반영했습니다. 감사합니다!

bh2980 committed:
fix(Company) : edit by code review

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+import CompanyPresenter from './Company';
+import { companyProfile, companyQuote } from './dummyData';
+
+const getCompanyProfile = () => {
```
굳이 함수 또 만들 필요 없을 것 같습니당

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+	return companyProfile[0];
+};
+
+const getCompanyQuote = () => {
```
이 친구도 마찬가지(불필요한 함수 선언)

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+};
+
+const Company = () => {
+	const [profile, setProfile] = useState(getCompanyProfile());
```
`useState(companyProfile[0])`

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+
+const Company = () => {
+	const [profile, setProfile] = useState(getCompanyProfile());
+	const [quote, setQuote] = useState(getCompanyQuote());
```
`useState(companyQuote[0])`

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+const Company = () => {
+	const [profile, setProfile] = useState(getCompanyProfile());
+	const [quote, setQuote] = useState(getCompanyQuote());
+
```
그리고 데이터 값이 useState의 set함수로 인해 바뀌는게 아니라면 상태관리 안쓰는게 좋아욥(불필요한 useState사용)

Climier-code commented on src/components/Company/CompanyContainer.jsx:
```diff
@@ -0,0 +1,20 @@
+	const [profile, setProfile] = useState(getCompanyProfile());
+	const [quote, setQuote] = useState(getCompanyQuote());
+
+	return <CompanyPresenter profile={profile} quote={quote} />;
```
그렇게 되면 밑에도 `<CompanyPresenter profile={companyProfile[0]} quote={companyQuote[0]} />` 로 축소 되겠죠

그러면 해당 Company 컴포넌트는 존재하는 의미는 딱히 없어집니다. CompanyPresenter에 있는 것들을 Company 컴포넌트로 넣는 것이 deps를 줄여줍니당

Climier-code commented on src/components/Company/StockChart.jsx:
```diff
@@ -0,0 +1,139 @@
+const StockChart = () => {
+	const [stockHistory, setStockHistory] = useState(getStockHistory(stockHistory1Day));
+
+	const chartChange = chartMode => {
```
해당 함수도 선언한 의미가 없어집니당

bh2980 replied:
위에 말씀해주신 내용들 다 포함해서 정리한 답 코멘트입니다.
제가 API 호출을 해본 적이 없어서 잘 몰라서 일단 밖에 선언해두긴 했는데 
API 넣기 시작하면 get~ 형태의 함수들이 API를 지속적으로 호출하는 함수가 될 것 같아요.
(찾아보니까 컴포넌트 안에 들어가더라구요...ㅎㅎ...)

그래서 get 함수 내부에는 호출 코드나 받은 데이터를 가공해서 return해주는 코드가 들어가고 
get 함수로 일정 시간마다 바뀐 가격과 가격 히스토리를 받아오게 될 것 같은데
혹시 이렇게 진행되는 경우에도 get함수랑 State를 지워야 할까요?

Company.jsx랑 컨테이너도 profile이랑 quote를 호출하는 부분을 
presenter랑 따로 작성해주려고 생각해서 분리해놓았는데
혹시 좋지 않은 방법일까요?

Climier-code replied:
만약에 API 콜이 계속 불러오게 된다면 함수+상태관리를 만들어두는게 맞으나 컴포넌트 밖이 아니라 해당 컴포넌트 안에서 선언하는 것이 더 적절하다고 생각합니다 ㅎㅎㅎ 컴포넌트 분리도 API적용 및 데이터 가공이 필요한 경우에는 잘 나눈 선택이라고 생각합니당!

bh2980 replied:
넵. 함수들은 안으로 넣어야겠네요. 감사합니다!

Climier-code commented on src/components/Company/StockChart.jsx:
```diff
@@ -0,0 +1,139 @@
+					<label
+						className="shadow-box"
+						htmlFor="oneday"
+						onClick={() => chartChange(stockHistory1Day)}
```
`onClick={() => setStockHistory(getStockHistory(stockHistory1Day))}`

밑에 것들도 맞게 적용해주세용

Climier-code commented on src/components/Company/Company.jsx:
```diff
@@ -11,74 +12,40 @@ const Company = () => {
-								alt="Apple Inc."
-								src="https://upload.wikimedia.org/wikipedia/commons/a/ab/Apple-logo.png"
-							></img>
+							<img alt={profile.companyName} src={profile.image}></img>
```
`<img ... />`

bh2980 replied:
얘는 계속 고쳤는데 계속 Undo하면서 사라지네요.  이럴 수가..이번에는 저장해야지...

Climier-code commented on src/components/Company/Company.jsx:
```diff
@@ -11,74 +12,40 @@ const Company = () => {
+								{new Intl.NumberFormat('en-US', {
+									style: 'currency',
+									currency: `${profile.currency}`,
+								}).format(`${quote.price}`)}
```
불필요한 템플릿 리터럴

`.format(quote.price)`

bh2980 replied:
수정했습니다.

Climier-code commented on src/components/Company/Company.jsx:
```diff
@@ -11,74 +12,40 @@ const Company = () => {
-								)}
+								{new Intl.NumberFormat('en-US', {
+									style: 'currency',
+									currency: `${profile.currency}`,
```
불필요한 템플릿 리터럴

`currency: profile.currency`

bh2980 replied:
수정했습니다.

Climier-code commented on src/components/Company/StockChart.jsx:
```diff
@@ -0,0 +1,139 @@
+		<div className="graph">
+			<ul>
+				<li>
+					<input type="radio" id="oneday" name="chart-date" value="1day" defaultChecked />
```
현재 상황은 radio input에 대한 이벤트가 아니라 label을 누를 때 이벤트가 발생되어 stockHistory가 변경되는 것 같은데 의도하신게 radio value state에 따른 차트 렌더링이면 label은 그대로 두고 radio에 대한 **상태** or **onClick** 이벤트가 되어야하지 않을까 싶습니당

ex) 1년에 해당되는 라벨을 눌렀을 경우 1년에 대한 차트는 나오지만 radio는 여전히 1일을 가리키는 상황

bh2980 replied:
일단 의도한 건 label을 누를 때 이벤트가 발생되어 stockHistory가 변경되는 것이 맞습니다. 
radio는 안 보이게 해놓고 label만 box형태로 구현해서 그렇게 생각했는데 
생각해보니 htmlFor 태그 있으니까 누르면 radio도 바뀌네요. 맞네........
안보이는데 radio를 왜 썼냐면 하나로 묶어두면 클릭 시 기존 체크 항목이 해제되니까 
단순하게 구현하기 편할 거라고 생각했거든요.
그런데 label이 바뀌는 걸 확인하고 올렸는데 안 바뀌나 보네요. 왜징...

radio value state로 차트 렌더링을 한다는 건 event.target.value같은 값을 받아서 
상태를 처리해주시는 방법을 말씀하시는 거 맞나요? 이 방법을 코멘트 달아주신 걸 읽고 떠올렸습니다.......

Climier-code replied:
아 이건 제 실수인 것 같습니댱...ㅎ... input의 id를 놓치고 말씀드렸었는데 잘 작동되네요 ㅎㅎㅎ 이쪽은 신경 안쓰셔도 될 것 같습니댱!

bh2980 replied:
앗 그래도 event로 하는 방법 알아갑니다...감사합니다!

bh2980 committed:
fix(Company) : edit by code review

remove normalize-url module
"useTabs": true, restore
deleted unused scss scroll code

bh2980 committed:
Merge branch 'develop' into feature/d650fe40
